src/crimson/osd/ops_executer.cc

```diff
@@ -451,20 +452,33 @@ static seastar::future<ceph::bufferlist> do_pgls_common(
     throw std::invalid_argument("outside of PG bounds");
   }
 
+  using entries_t = decltype(pg_ls_response_t::entries);
   return backend.list_objects(lower_bound, limit).then(
-    [pg_end, nspace, &backend](auto objects, auto next) {
+    [&backend, filter, nspace](auto objects, auto next) {
+      return seastar::when_all_succeed(
+        seastar::map_reduce(std::move(objects),
+	  [&backend, filter, nspace](const hobject_t& obj) {
+	    if (obj.get_namespace() == nspace) {
+	      if (filter) {
+	        return pgls_filter(*filter, backend, obj);
+	      } else {
+	        return seastar::make_ready_future<hobject_t>(obj);
+	      }
+	    } else {
+	      return seastar::make_ready_future<hobject_t>(hobject_t{});
+	    }
+	  }, entries_t{},
```

Comment:
nit: put `entries_t{}` in dedicated line to visually reflect it's an argument for `map_reduce()`?

Reply:
updated and repushed.